tests/cpp/core/RestWindowsTests.cpp

```diff
@@ -87,20 +86,34 @@ BOOST_AUTO_TEST_CASE( testWindowInfo )
 
     auto future = windowsContent.getWindowInfo( std::string(), std::string( ));
     auto response = future.get();
-    BOOST_CHECK( response.code ==  200 );
+    BOOST_CHECK_EQUAL( response.code, 200 );
 
     const auto uuid = window->getID().toString().replace( _regex, "" );
 
+    future = windowsContent.getWindowInfo( uuid.toStdString() + "/thumbnail",
+                                           std::string( ));
+    response = future.get();
+    BOOST_CHECK_EQUAL( response.code, 204 );
+
+    // Wait for the thumnbnail generation
     sleep(2);
+    future = windowsContent.getWindowInfo( uuid.toStdString() + "/thumbnail",
+                                           std::string( ));
+    response = future.get();
+    BOOST_CHECK_EQUAL( response.payload,_getThumbnail( ));
+    BOOST_CHECK_EQUAL( response.code, 200 );
+    BOOST_CHECK_EQUAL( response.headers[zeroeq::http::Header::CONTENT_TYPE],
+                       "image/png" );
+
+    displayGroup->removeContentWindow( window );
 
     future = windowsContent.getWindowInfo( uuid.toStdString() + "/thumbnail",
                                            std::string( ));
     response = future.get();
-    BOOST_CHECK( response.payload ==  _getThumbnail() );
-    BOOST_CHECK( response.code ==  200 );
+    BOOST_CHECK_EQUAL( response.code, 204 );
```

Comment:
shouldn't it be a 404 for a window that has been removed? Otherwise the client will keep asking for it?